tests/build-rss.test.js

```diff
@@ -0,0 +1,101 @@
+const fs = require('fs');
+const path = require('path');
+const rssFeed = require('../scripts/build-rss');
+const { mockRssData, title, type, desc, outputPath } = require('./fixtures/rssData');
+
+jest.mock('../config/posts.json', () => mockRssData, { virtual: true });
+
+describe('rssFeed', () => {
+  const testOutputDir = path.join(__dirname, '..', 'public', 'test-output');
+
+  beforeEach(() => {
+    fs.mkdirSync(testOutputDir, { recursive: true });
+  });
+
+  afterEach(() => {
+    if (fs.existsSync(testOutputDir)) {
+      fs.readdirSync(testOutputDir).forEach(file => {
+        fs.unlinkSync(path.join(testOutputDir, file));
+      });
+      fs.rmdirSync(testOutputDir);
+    }
+  });
+
+  it('should generate RSS feed and write to file', () => {
+    try {
+      rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      throw new Error('Error encountered during test execution:', err)
+    }
+
+    const filePath = path.join(__dirname, '..', 'public', outputPath);
+    expect(fs.existsSync(filePath)).toBe(true);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+    expect(fileContent).toContain('<rss version="2.0"');
+    expect(fileContent).toContain('<title>Test Blog RSS</title>');
+  });
+
+  it('should prioritize featured posts over non-featured ones', () => {
+    try {
+      rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      throw new Error('Error encountered during test execution:', err)
+    }
+
+    const filePath = path.join(__dirname, '..', 'public', outputPath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+
+    const itemTitles = fileContent.match(/<title>(.*?)<\/title>/g);
+
+    expect(itemTitles[1]).toContain('Test Post 1');
+    expect(itemTitles[2]).toContain('Another Featured Post');
+    expect(itemTitles[3]).toContain('Non-Featured Post 1');
+  });
+
+  it('should sort posts by date in descending order', () => {
+    try {
+      rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      throw new Error('Error encountered during test execution:', err)
```

Comment:
Why error has been thrown in the test execution, it will then lead to stop the execution of further tests in the script. You shouldn't throw error in tests.

Reply:
You mentioned it in comment did i get it wrong 😅 
https://github.com/asyncapi/website/pull/3101#discussion_r1739322472

Reply:
Replaced it with console.error